internal/core/writer/properties.go

```diff
@@ -76,7 +76,7 @@ func newDimensionPropertyClient(conf *config.WriterConfig) (*dimensionPropertyCl
 func (dpc *dimensionPropertyClient) SetPropertiesOnDimension(dimProps *types.DimProperties) error {
 	filteredDimProps := &(*dimProps)
 
-	filteredDimProps = dpc.PropertyFilterSet.FilterDimProps(filteredDimProps)
+	filteredDimProps = dpc.PropertyFilterSet.FilterDimProps(filteredDimProps).Copy()
```

Comment:
This will blow up if `filteredDimProps` is nil, but I don't think it can be nil at this point because of how `FilterDimProps` is implemented.  So I would remove the nil check below this line to make it less confusing about the potential states of `filteredDimProps`.